chalk/shape.py

```diff
@@ -242,28 +243,44 @@ class Arc(Shape):
     radius: float
     angle0: float
     angle1: float
-
     def __post_init__(self) -> None:
         surface = cairo.SVGSurface("undefined.svg", 1280, 200)
         self.ctx = cairo.Context(surface)
 
     def get_bounding_box(self) -> BoundingBox:
         self.render(self.ctx)
         l, t, r, b = self.ctx.path_extents()
-        return BoundingBox([Point(l, t), Point(r, b)])
+        return BoundingBox([P2(l, t), P2(r, b)])
+
 
+    def get_envelope(self) -> Envelope:
```

Comment:
I've noticed in the Big Ben tutorial that the `under_arc` curve is chopped off when rendered. On closer inspection, it seems that there is a slight issue with the envelope. An arc can, in principle, be specified by negative angles:
```python
 arc(radius=1, angle0=-50, angle1=50)
```
but in the `wrapped` function the direction `d` takes only positive angles; so if a direction's angle is in the `[310, ..., 360]` range then the `if` condition enters through the second branch ("point outside of arc"). For the above example, the envelope looks like this (see the upper half of the envelope, which corresponds to the `[-50, ..., 0]` range):
![o](https://user-images.githubusercontent.com/819256/177218187-1226c8c0-3519-4c6c-9358-1fc165ad814a.png)